File: src/strings/and/arrays/IntegerToRoman.java
package strings.and.arrays;

import java.util.Arrays;
import java.util.concurrent.atomic.AtomicInteger;

public class IntegerToRoman {

    public static void main(String[] args) {
    IntegerToRoman obj = new IntegerToRoman();
        System.out.println(obj.intToRoman(4));
    }

    public String intToRoman(int num) {

        StringBuilder ans = new StringBuilder();
        final String[] ROMAN = new String[]{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        final int[] INTEGER = new int[]{1000,900,500,400,100,90,50,40,10,9,5,4,1};
        for(int i=0; i<INTEGER.length;i++){
            while (num>=INTEGER[i]){
                num-=INTEGER[i];
                ans.append(ROMAN[i]);
            }
        }

        return String.valueOf(ans);
    }
}

File: bin/strings/and/arrays/js/IntegerToRoman.js
/**
 * @param {number} num
 * @return {string}
 */
var intToRoman = function(num) {
    const ROMAN = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX","V", "IV", "I"]
    const INTEGER = [1000,900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
    let ans = ""
    INTEGER.forEach((element, index) =>{

        while(num>=element){
            num-=element
            ans+=(ROMAN[index])
        }
    })

    return ans

};